### PR TITLE
Adds Economy Inflation random event

### PR DESCRIPTION
**Pull request by carlarctg**
# About the pull request

Adds a new (and currently the only) random event, 'Economy Inflation'. The USS Almayer recieves a broad-band signal sent to all nearby UA-affiliated colonies and vessels that requests a specific kind of vendors stationed on them have their prices increased, due to some randomly-selected incident backstory. There will be a random delay before the change comes into effect, so make sure you buy all you need beforehand! The announcement will only be visible shipside to avoid communications clutter.

Added 'product type' variable to non-CM vendors, such as food, beverages, souto, recreation, used for the above. 

Added a global list for non-CM vending machines.

# Explain why it's good for the game

Random events are a great way to spice up the round by introducing new, non-player-created variables that don't heavily impact the round and allow players to utilize them to create new RP or gameplay avenues. Such as a marine desperately buying out every Souto vendor before the prices increase by 400%, complaining on communications about the 'dang CLF taking over all those colonies' or 'the gosh darn economy these days', and so forth. 

This PR also introduces an actual random event to use as template for single celled contributor organisms to be able to copypaste it and make a new event that way.

# Testing Photographs and Procedure

<details>
<summary>Screenshots & Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

# Changelog

:cl:
add: Adds a new (and currently the only) random event, 'Economy Inflation'. The USS Almayer recieves a broad-band signal sent to all nearby UA-affiliated colonies and vessels that requests a specific kind of vendors stationed on them have their prices increased, due to some randomly-selected incident backstory. There will be a random delay before the change comes into effect, so make sure you buy all you need beforehand! The announcement will only be visible shipside to avoid communications clutter.
code: Added 'product type' variable to non-CM vendors, such as food, beverages, souto, recreation, used for the above. 
code: Added a global list for non-CM vending machines.
/:cl:

### CODE REVIEW

File: code/game/machinery/vending/vending.dm [hunk 1]
```diff
@@ -87,6 +93,10 @@
 	power_change()
 	start_processing()
 
+/obj/structure/machinery/vending/Destroy()
+	LAZYREMOVE(GLOB.total_vending_machines, src)
+	. = ..()
```
Comment (Zonespace27) [inline code/game/machinery/vending/vending.dm:98]:
```suggestion
	return ..()
```

File: code/modules/events/inflation.dm [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+	min_players = 1
+	max_occurrences = 3 // any more and it'd get tiresome
+	alert_observers = FALSE
+	gamemode_blacklist = list("Whiskey Outpost", "Hive Wars")
+
+/datum/round_event/economy_inflation
+	announce_when = 1
+	startWhen = 10
+	endWhen = 15
+	var/product_type = VENDOR_PRODUCT_TYPE_FOOD
```
Comment (Zonespace27) [inline code/modules/events/inflation.dm:16]: codedoc these

File: code/modules/events/inflation.dm [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+		"*garbled static*",
+		"flaring tensions with Arcturus",
+		"a Sol-wide solar flare technological blackout",
+		"\[REDACTED\]",
+		"the release of a competitor's product",
+	)
+	return pick(backstories)
+
+/datum/round_event/economy_inflation/proc/get_percentage()
+	return (inflation_mult -1) * 100 // example result: 3.5 -> 250%
```
Comment (Zonespace27) [inline code/modules/events/inflation.dm:76]:
```suggestion
	return (inflation_mult - 1) * 100 // example result: 3.5 -> 250%
```

File: code/modules/events/inflation.dm [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+	for(var/obj/structure/machinery/vending/vending_machine as anything in GLOB.total_vending_machines)
+		if(!is_mainship_level(vending_machine.z) || vending_machine.product_type != product_type)
+			continue
+		vending_machine.speak("Prices have been updated!")
+		playsound(vending_machine, 'sound/machines/twobeep.ogg', 25, TRUE)
+		var/display_records = vending_machine.product_records + vending_machine.hidden_records + vending_machine.coin_records
+		for(var/datum/data/vending_product/product in display_records)
+			product.price *= inflation_mult
+
+/datum/round_event/economy_inflation/proc/get_random_story()
```
Comment (Zonespace27) [inline code/modules/events/inflation.dm:45]: codedoc

File: code/modules/events/inflation.dm [hunk 5]
```diff
@@ -0,0 +1,76 @@
+
+/datum/round_event_control/economy_inflation
```
Comment (Zonespace27) [inline code/modules/events/inflation.dm:2]:
```suggestion
/datum/round_event_control/economy_inflation
```

**Review by Zonespace27 — CHANGES_REQUESTED** [review]

File: code/modules/events/inflation.dm [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+	name = "Economy Inflation"
+	typepath = /datum/round_event/economy_inflation
+	weight = 10 // probably could use some defines in the future - EVENT_UNLIKELY, EVENT_COMMON, etc.
+	earliest_start = 5 MINUTES
+	min_players = 1
+	max_occurrences = 3 // any more and it'd get tiresome
```
Comment (stanalbatross) [inline code/modules/events/inflation.dm:7]:
```suggestion
	max_occurrences = 1 // any more and it'd get tiresome
```

File: code/modules/events/inflation.dm [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+		"public leaks of corporate secrets",
+		"a high-profile assassination",
+		"a CLF take-over of an important industrial colony",
+		"discovery of high levels of an unhealthy chemical",
+		"a public scandal involving a corporate chairman and a farm animal",
+		"large-scale colony riots",
+		"suppressed factory riots over riot responses",
+		"CLF bombing of vital industrial colonies",
+		"a tense stand-off on several mining stations with UPP insurgents",
+		"a publicized Xenomorph infestation",
```
Comment (stanalbatross) [inline code/modules/events/inflation.dm:62]:
```suggestion
```
LRP they would not publicise this

File: code/modules/events/inflation.dm [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+		"a large sector of colonized space going dark for unknown reasons", //XX-121,
+		"withdrawal of government funding in relevant sectors",
+		"a vital Seegson industries space station going dark", // A:I,
+		"several mining colonies defecting to the UPP",
+		"a critical USCM defeat by CLF insurgents in the Tychon's Rift sector",
+		"newly imposed sanctions as a result of corporate investigations",
+		"*garbled static*",
+		"flaring tensions with Arcturus",
+		"a Sol-wide solar flare technological blackout",
+		"\[REDACTED\]",
```
Comment (stanalbatross) [inline code/modules/events/inflation.dm:75]:
```suggestion
```
we do not use lazy scp writing style here

File: code/modules/events/inflation.dm [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+		"a succession of hostile takeovers by the Company",
+		"widespread corporate bankruptcy",
+		"pirate takeover of an essential cargo fleet",
+		"a large sector of colonized space going dark for unknown reasons", //XX-121,
+		"withdrawal of government funding in relevant sectors",
+		"a vital Seegson industries space station going dark", // A:I,
+		"several mining colonies defecting to the UPP",
+		"a critical USCM defeat by CLF insurgents in the Tychon's Rift sector",
+		"newly imposed sanctions as a result of corporate investigations",
+		"*garbled static*",
```
Comment (stanalbatross) [inline code/modules/events/inflation.dm:72]:
```suggestion
```
as above

File: code/modules/events/inflation.dm [hunk 10]
```diff
@@ -0,0 +1,81 @@
...
+		"a high-profile assassination",
+		"a CLF take-over of an important industrial colony",
+		"discovery of high levels of an unhealthy chemical",
+		"a public scandal involving a corporate chairman and a farm animal",
+		"large-scale colony riots",
+		"suppressed factory riots over riot responses",
+		"CLF bombing of vital industrial colonies",
+		"a tense stand-off on several mining stations with UPP insurgents",
+		"a publicized Xenomorph infestation",
+		"a succession of hostile takeovers by the Company",
```
Comment (stanalbatross) [inline code/modules/events/inflation.dm:63]: what company?? and why would this cause price increases

**Review by stanalbatross — CHANGES_REQUESTED** [review]

File: code/modules/events/inflation.dm [hunk 11]
```diff
@@ -0,0 +1,81 @@
+/datum/round_event_control/economy_inflation
+	name = "Economy Inflation"
+	typepath = /datum/round_event/economy_inflation
+	weight = 10 // probably could use some defines in the future - EVENT_UNLIKELY, EVENT_COMMON, etc.
+	earliest_start = 5 MINUTES
+	min_players = 1
+	max_occurrences = 3 // any more and it'd get tiresome
+	alert_observers = FALSE
+	gamemode_blacklist = list("Whiskey Outpost", "Hive Wars")
```
[the remark this answers is not in the document]
Comment (carlarctg) [reply code/modules/events/inflation.dm:9]: yeah okay let me touch 30000 files and get 30000 merge econflicts and turn these into a define okay duge

File: code/modules/events/inflation.dm (hunk 6 above)
Comment (carlarctg) [reply code/modules/events/inflation.dm:7]: riiiiiiiiiiiiiiiiiiight

File: code/modules/events/inflation.dm (hunk 7 above)
Comment (carlarctg) [reply code/modules/events/inflation.dm:62]: leaked by civilians

File: code/modules/events/inflation.dm (hunk 8 above)
Comment (carlarctg) [reply code/modules/events/inflation.dm:75]: literally completely out of context and has nothing to do with the line here - [REDACTED] in scp articles is used as a lazy and cheap way to let the reader fill in the blanks without needing the writer to put any effort in, this is a one out of twenty chance backstory that lets the players talk about what could possibly be redacted

File: code/modules/events/inflation.dm (hunk 10 above)
Comment (carlarctg) [reply code/modules/events/inflation.dm:63]: W-Y? everyone calls it the Company its almost like youre not an aliens fan

Comment (carlarctg) [reply code/modules/events/inflation.dm:63]: hostile takeovers would likely cause price increases if people worry about what that could mean

File: code/modules/events/nothing_happens.dm [hunk 12]
```diff
@@ -0,0 +1,12 @@
+
+// Due to our blatant lack of events, there is no real better solution that isn't worse shitcode.
```
Comment (Zonespace27) [inline code/modules/events/nothing_happens.dm:2]: (This is entirely optional, but would be very nice)

You could, on Initialize() of this event, subtract the weight of every event from the base weight of 150, so nobody needs to update it ever, until we have enough events to remove this entirely

File: code/modules/events/nothing_happens.dm [hunk 13]
```diff
@@ -10,3 +9,11 @@
 	min_players = 1
 	max_occurrences = INFINITE_EVENT_OCURRENCES
 	alert_observers = FALSE
+
+/datum/round_event_control/nothing_happens/New()
+	. = ..()
+	for(var/datum/round_event_control/possible_event in subtypesof(/datum/round_event_control))
+		if(possible_event.type == src.type)
+			break
+		src.weight -= possible_event.weight
```
Comment (Zonespace27) [inline code/modules/events/nothing_happens.dm:18]: this won't work and it's obvious this is untested

`subtypesof()` gets a list of paths, so for loop will iterate exactly 0 times (add `as anything`)
additionally, you're trying to get a var on a path without using initial(), which is also necessary
finally, remove the `src.`

Comment (carlarctg) [reply code/modules/events/nothing_happens.dm:18]: SAD

**Review by harryob — APPROVED** [review]

**Pull request merged**